hpat/tests/tests_perf/test_perf_unicode.py

```diff
@@ -0,0 +1,145 @@
+
+        iter_time.append(local_result)
+
+    return iter_time
+
+
+def usecase_join(input_data):
+    test_iteration_number = 5
+    iter_time = []
+
+    for local_iter in range(test_iteration_number):
+        start_time = time.time()
+
+        for string in input_data:
+            'X'.join([string, string, string, string])
+
+        finish_time = time.time()
+        local_result = finish_time - start_time
+
+        iter_time.append(local_result)
+
+    return iter_time
+
+
+def usecase_center(input_data):
+    result_str_grow_factor = 1.4
+    test_iteration_number = 5
+    iter_time = []
+
+    for local_iter in range(test_iteration_number):
+        start_time = time.time()
+
+        for string in input_data:
+            # The width must be an Integer
+            new_string_len = int(len(string) * result_str_grow_factor)
+
+            string.center(new_string_len, '+')
+
+        finish_time = time.time()
+        local_result = finish_time - start_time
+
+        iter_time.append(local_result)
+
+    return iter_time
+
+
+class TestStringMethods(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.test_results = TestResults()
+        cls.test_results.load()
+
+        cls.total_data_size_bytes = [1.0E+07]
+        cls.width = [16, 64, 512, 1024]
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.test_results.print()
+
+    def test_unicode_split(self):
+        pyfunc = usecase_split
+        hpat_func = numba.njit(pyfunc)
+
+        for data_size in self.total_data_size_bytes:
+            for data_width in self.width:
+                test_data = perf_data_gen(STRIP_CASES, data_width, data_size)
+                self.test_results.add('unicode_split', 'JIT', len(test_data), data_width, hpat_func(test_data))
+                self.test_results.add('unicode_split', 'Reference', len(test_data), data_width, pyfunc(test_data))
+
+    def test_unicode_join(self):
+        pyfunc = usecase_join
+        hpat_func = numba.njit(pyfunc)
+
+        for data_size in self.total_data_size_bytes:
+            for data_width in self.width:
+                test_data = perf_data_gen(STRIP_CASES, data_width, data_size)
+                self.test_results.add('unicode_join', 'JIT', len(test_data), data_width, hpat_func(test_data))
+                self.test_results.add('unicode_join', 'Reference', len(test_data), data_width, pyfunc(test_data))
+
+    def test_unicode_center(self):
+        pyfunc = usecase_center
+        hpat_func = numba.njit(pyfunc, parallel=True)
+
+        for data_size in self.total_data_size_bytes:
+            for data_width in self.width:
+                test_data = perf_data_gen(STRIP_CASES, data_width, data_size)
+                self.test_results.add('unicode_center', 'JIT', len(test_data), data_width, hpat_func(test_data))
+                self.test_results.add('unicode_center', 'Reference', len(test_data), data_width, pyfunc(test_data))
+
+
+if __name__ == "__main__":
```

Comment:
I think we need to provide an ability to run tests in particular file by executing it.
For this we need to add PYTHONPATH and do some other things.
I did it locally and provide an example later.

Reply:
Ok, please provide the example.